vars/runGradleCheck.groovy

```diff
@@ -0,0 +1,59 @@
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def git_repo_url = args.gitRepoUrl ?: 'null'
+    def git_reference = args.gitReference ?: 'null'
+
+    println("Git Repo: ${git_repo_url}")
+    println("Git Reference: ${git_reference}")
+
+    if (git_repo_url.equals('null') || git_reference.equals('null')) {
+        println("No git repo url or git reference to checkout the commit, exit 1")
+        System.exit(1)
+    }
+    else {
+        withCredentials([
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-credentials", usernameVariable: 'amazon_s3_access_key', passwordVariable: 'amazon_s3_secret_key'),
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-resources", usernameVariable: 'amazon_s3_base_path', passwordVariable: 'amazon_s3_bucket'),
+            usernamePassword(credentialsId: "jenkins-github-bot-token", usernameVariable: 'GITHUB_BOT_USER', passwordVariable: 'GITHUB_BOT_TOKEN')]) {
+
+            sh """
+
+                set -e
+                set +x
+
+                env | grep JAVA | grep HOME
+
+                echo "Git clone: ${git_repo_url} with ref: ${git_reference}"
+                rm -rf search
+                git clone ${git_repo_url} search
+                cd search/
+                git checkout -f ${git_reference}
+                git rev-parse HEAD
```

Comment:
Why not use git groovy scripts here? Example https://github.com/opensearch-project/opensearch-build/blob/09f65b34a719581a40397a58b081b4dafc12ce5e/jenkins/opensearch-ruby/Jenkinsfile#L30

Reply:
I have more control over here.
It is just a preference as majority of this is shell script.
So no need to use a jenkins specific groovy lib to do so.